include/alpaka/mem/buf/oacc/Copy.hpp

```diff
@@ -0,0 +1,395 @@
+                ALPAKA_FN_HOST auto operator()() const
+                -> void
+                {
+                    ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                    this->printDebug();
+#endif
+                    Vec<DimMin1, ExtentSize> const extentWithoutInnermost(subVecBegin<DimMin1>(this->m_extent));
+                    // [z, y, x] -> [y, x] because the z pitch (the full size of the buffer) is not required.
+                    Vec<DimMin1, DstSize> const dstPitchBytesWithoutOutmost(subVecEnd<DimMin1>(this->m_dstPitchBytes));
+                    Vec<DimMin1, SrcSize> const srcPitchBytesWithoutOutmost(subVecEnd<DimMin1>(this->m_srcPitchBytes));
+
+                    if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
+                    {
+                        this->m_dev.makeCurrent();
+                        meta::ndLoopIncIdx(
+                            extentWithoutInnermost,
+                            [&](Vec<DimMin1, ExtentSize> const & idx)
+                            {
+                                this->m_copyPred(
+                                    reinterpret_cast<void *>(
+                                        this->m_dstMemNative + (castVec<DstSize>(idx) * dstPitchBytesWithoutOutmost).foldrAll(std::plus<DstSize>())),
+                                    const_cast<void*>(reinterpret_cast<const void *>(
+                                        this->m_srcMemNative + (castVec<SrcSize>(idx) * srcPitchBytesWithoutOutmost).foldrAll(std::plus<SrcSize>()))),
+                                    static_cast<std::size_t>(this->m_extentWidthBytes));
+                            });
+                    }
+                }
+            };
+
+            //#############################################################################
+            //! The 1d Oacc memory copy task.
+            template<
+                typename TViewDst,
+                typename TViewSrc,
+                typename TExtent,
+                typename TCopyPred>
+            struct TaskCopyOacc<
+                DimInt<1>,
+                TViewDst,
+                TViewSrc,
+                TExtent,
+                TCopyPred>
+                    : public TaskCopyOaccBase<DimInt<1>, TViewDst, TViewSrc, TExtent, TCopyPred>
+            {
+                //-----------------------------------------------------------------------------
+                using TaskCopyOaccBase<DimInt<1u>, TViewDst, TViewSrc, TExtent, TCopyPred>::TaskCopyOaccBase;
+
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST auto operator()() const
+                -> void
+                {
+                    ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                    this->printDebug();
+#endif
+                    if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
```

Comment:
`if(this->m_extent.prod() != 0)` should be enough.

Reply:
Sometimes (in some tests or examples), extents are signed, causing a warning without the cast.

Reply:
That's why I would use `0` and not `0u`, as I wrote.

Reply:
Aha, I did not notice that.

Would this not also raise a singe compare warning in the opposite case, i.e. when extent is unsigned?

Reply:
It should not, because the compiler can see that the int literal 0 does not change it's value when promoted to an unsigned integral. But you are free to proof me wrong! ;)

Reply:
First I would need to figure out which compiler warns about
```c++
if(this->m_extent.prod() != 0u)
{}
```
when extent is signed. GCC 10.2 does not.

Even just removing the `static_cast` may compile without warnings with most compilers, but there was at least one in the CI where it failed, which is why the cast is there. I would guess, that a compiler which warns with `0u` would also warn with `0`, because it does not have the logic see that the sign does not matter when comparing to 0.

I would have to run a changed version through the CI again to be sure and know which compiler cause this line to get a cast, which I am reluctant to do.

Reply:
Ok, the CI ran through on this one, so I will leave it your way. Not sure which compiler it failed on in the past, or if I did this preemptively after all.